master/internal/db/postgres_checkpoints.go

```diff
@@ -148,20 +148,15 @@ func UpdateCheckpointSizeTx(ctx context.Context, idb bun.IDB, checkpoints []uuid
 	var experimentIDs []int
 	err := idb.NewRaw(`
 UPDATE trials SET checkpoint_size=sub.size, checkpoint_count=sub.count FROM (
-	SELECT trial_id, sum(size) as size, sum(count) as count
-	FROM (
-		WITH trial_ids AS (
-			SELECT t.id AS trial_id
-			FROM checkpoints_v2 INNER JOIN trials t ON checkpoints_v2.task_id = t.task_id
-			WHERE uuid IN (?)
-		)
-		SELECT t.id AS trial_id,
-		COALESCE(SUM(size) FILTER (WHERE checkpoints_v2.state != 'DELETED'), 0) AS size,
-		COUNT(*) FILTER (WHERE checkpoints_v2.state != 'DELETED') AS count
-		FROM checkpoints_v2 INNER JOIN trials t on checkpoints_v2.task_id = t.task_id
-		WHERE t.id IN (SELECT trial_id FROM trial_ids)
-		GROUP BY t.id
-	) ssub
+	SELECT trial_id,
```

Comment:
Changed the query to avoid using the view. 

```
EXPLAIN ANALYZE UPDATE trials SET checkpoint_size=sub.size, checkpoint_count=sub.count FROM (
    SELECT
        trial_id,
        COALESCE(SUM(size) FILTER (WHERE state != 'DELETED'), 0) AS size,
        COUNT(*) FILTER (WHERE state != 'DELETED') AS count
    FROM checkpoints_v2
    JOIN trial_id_task_id tt ON tt.task_id = checkpoints_v2.task_id
    WHERE
        trial_id IN (
            SELECT tt.trial_id FROM checkpoints_v2
            LEFT JOIN trial_id_task_id tt ON tt.task_id = checkpoints_v2.task_id
            WHERE uuid IN ('00015956-9958-4e19-8fbc-69242cb8a934', '000261b0-a0b5-4094-9afe-80f4f5748be8', '000371c9-02dc-4dbb-9f1b-21d80e931257', '000576fa-a7f7-48cc-ba82-3791b0328847', '000603ab-06c0-49e5-9a0e-44d4858be4e5')
        )
    GROUP BY trial_id
) sub
WHERE trials.id = sub.trial_id
RETURNING experiment_id
```

```
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-----------------------------------------------------------------------
 Update on trials  (cost=95.12..162.02 rows=8 width=987) (actual time=0.419..0.550 rows=5 loops=1)
   ->  Nested Loop  (cost=95.12..162.02 rows=8 width=987) (actual time=0.376..0.454 rows=5 loops=1)
         ->  Subquery Scan on sub  (cost=94.83..95.13 rows=8 width=112) (actual time=0.357..0.393 rows=5 loops=1)
               ->  GroupAggregate  (cost=94.83..95.05 rows=8 width=44) (actual time=0.351..0.382 rows=5 loops=1)
                     Group Key: tt.trial_id
                     ->  Sort  (cost=94.83..94.85 rows=8 width=16) (actual time=0.341..0.348 rows=34 loops=1)
                           Sort Key: tt.trial_id
                           Sort Method: quicksort  Memory: 27kB
                           ->  Nested Loop  (cost=84.40..94.71 rows=8 width=16) (actual time=0.171..0.321 rows=34 loops=1)
                                 ->  Nested Loop  (cost=83.99..85.53 rows=5 width=26) (actual time=0.147..0.175 rows=5 loops=1)
                                       ->  HashAggregate  (cost=83.70..83.75 rows=5 width=4) (actual time=0.139..0.143 rows=5 loops=1)
                                             Group Key: tt_1.trial_id
                                             ->  Nested Loop  (cost=0.70..83.69 rows=5 width=4) (actual time=0.048..0.132 rows=5 loops=1)
                                                   ->  Index Scan using checkpoints_v2_uuid_key on checkpoints_v2 checkpoints_v2_1  (cost=0.41..42.16 rows=5 width=23) (actual time=0.021..0.043 rows=5 loops=1)
                                                         Index Cond: (uuid = ANY ('{00015956-9958-4e19-8fbc-69242cb8a934,000261b0-a0b5-4094-9afe-80f4f5748be8,000371c9-02dc-4dbb-9f1b-21d80e931257,000576fa-a7f7-4
8cc-ba82-3791b0328847,000603ab-06c0-49e5-9a0e-44d4858be4e5}'::uuid[]))
                                                   ->  Index Scan using trial_id_task_id_task_id_key on trial_id_task_id tt_1  (cost=0.29..8.30 rows=1 width=26) (actual time=0.016..0.016 rows=1 loops=5)
                                                         Index Cond: (task_id = checkpoints_v2_1.task_id)
                                       ->  Index Scan using idx_trial_id_task_id_run_id on trial_id_task_id tt  (cost=0.29..0.35 rows=1 width=26) (actual time=0.005..0.005 rows=1 loops=5)
                                             Index Cond: (trial_id = tt_1.trial_id)
                                 ->  Index Scan using ix_checkpoints_v2_task_id_steps_completed on checkpoints_v2  (cost=0.41..1.81 rows=2 width=35) (actual time=0.017..0.025 rows=7 loops=5)
                                       Index Cond: (task_id = tt.task_id)
         ->  Index Scan using trials_pkey on trials  (cost=0.29..8.36 rows=1 width=863) (actual time=0.009..0.009 rows=1 loops=5)
               Index Cond: (id = sub.trial_id)
 Planning time: 2.140 ms
 Execution time: 0.724 ms
(25 rows)
```